crates/aggchain-proof-core/src/proof.rs

```diff
@@ -44,31 +38,11 @@ pub struct AggchainProofWitness {
 impl AggchainProofWitness {
     pub fn verify_aggchain_inputs(&self) -> Result<AggchainProofPublicValues, ProofError> {
         // Verify the FEP proof or ECDSA signature.
-        self.fep.verify()?;
-
-        // Verify that the `l1Head` considered by the FEP exists in the L1 Info Tree
-        {
-            if self.fep.l1_head != self.l1_info_tree_leaf.1.block_hash {
-                return Err(ProofError::MismatchL1Head {
-                    from_l1_info_tree_leaf: self.l1_info_tree_leaf.1.block_hash,
-                    from_fep_public_values: self.fep.l1_head,
-                });
-            }
-
-            let inclusion_proof_valid = self.l1_head_inclusion_proof.verify(
-                self.l1_info_tree_leaf.1.hash(),
-                self.l1_info_tree_leaf.0,
-                self.l1_info_root,
-            );
-
-            if !inclusion_proof_valid {
-                return Err(ProofError::InvalidInclusionProofL1Head {
-                    index: self.l1_info_tree_leaf.0,
-                    l1_leaf_hash: self.l1_info_tree_leaf.1.hash(),
-                    l1_info_root: self.l1_info_root,
-                });
-            }
-        }
+        self.fep.verify(
+            self.l1_info_root,
+            self.new_local_exit_root,
+            compute_commit_imported_bridge_exits(&self.global_indices),
```

Comment:
maybe we can avoid computing this twice by having this computation once in the high level function and propagating its value where needed?